example/lib/pages/connect.dart

```diff
@@ -93,6 +103,13 @@ class _ConnectPageState extends State<ConnectPage> {
 
       // Create a Listener before connecting
       final listener = room.createListener();
+      E2EEOptions? e2eeOptions;
+      if (_e2ee) {
+        final keyProvider = await BaseKeyProvider.create(sharedKey: true);
```

Comment:
maybe our default implementation doesn't even need `sharedKey` to be configurable, but rather have it set in the constructor.  

that would allow us to _only_ have `keyProvider.setKey(myKey)` as an API for the default implementation and users could create their own more complex ones by inheriting from the base. 
What I'm suggesting is having `ExternalKeyprovider` additionally that inherits from `BaseKeyProvider` but only exposes a single `setKey` method.